Sources/AsyncProcess/ProcessExecutor.swift

```diff
@@ -486,7 +486,7 @@ public final actor ProcessExecutor {
       }
       if p.isRunning {
         self.logger.info("terminating process", metadata: ["pid": "\(childPid)"])
-        kill(childPid, SIGKILL)
+        kill(-childPid, SIGINT)
```

Comment:
I wouldn't recommend that. Structured Concurrency requires that the process _will definitely exit_ or else we're just gonna deadlock. But I think we can update `ProcessExecutor` to be able to run a shutdown sequence instead of just sending SIGKILL

Reply:
Sorry, does "process" mean spawned child process, or the SDK generator process itself? I don't think `kill(2)` does not block the sender process.

Reply:
`kill` doesn't block. But let's assume that the child doesn't exit if you send a `kill(SIGINT, child)`, then we'll potentially wait for that child forever. So it's very important that (at some point) we do send a `SIGKILL` if the child doesn't disappear by itself.

But see #100 for teardown sequences which should fix this

Reply:
Ah, that sounds reasonable. Thanks!